### PR TITLE
CORE-10678 Merge beta 2 branch into release/os/5

### PR DESCRIPTION
**Pull request by williamvigorr3**
```
git merge origin/release/os/5.0-Beta2
Auto-merging charts/corda/Chart.yaml
Auto-merging components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberResourceClientImpl.kt
Auto-merging components/membership/membership-persistence-client-impl/src/main/kotlin/net/corda/membership/impl/persistence/client/MembershipPersistenceClientImpl.kt
Auto-merging components/membership/membership-persistence-client-impl/src/test/kotlin/net/corda/membership/impl/persistence/client/MembershipPersistenceClientImplTest.kt
Auto-merging components/membership/membership-persistence-client/src/main/kotlin/net/corda/membership/persistence/client/MembershipPersistenceClient.kt
Auto-merging components/membership/membership-persistence-service-impl/src/integrationTest/kotlin/net/corda/membership/impl/persistence/service/MembershipPersistenceTest.kt
Auto-merging components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandler.kt
CONFLICT (content): Merge conflict in components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandler.kt
Auto-merging components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/MembershipPersistenceRPCProcessorTest.kt
Auto-merging components/membership/membership-service-impl/src/main/kotlin/net/corda/membership/service/impl/MemberOpsServiceProcessor.kt
Auto-merging components/membership/membership-service-impl/src/test/kotlin/net/corda/membership/service/impl/MemberOpsServiceProcessorTest.kt
Auto-merging components/membership/registration-impl/src/integrationTest/kotlin/net/corda/membership/impl/registration/dummy/TestMembershipPersistenceClientImpl.kt
Auto-merging components/membership/synchronisation-impl/src/integrationTest/kotlin/net/corda/membership/impl/synchronisation/dummy/TestMembershipPersistenceClient.kt
Auto-merging components/membership/synchronisation-impl/src/main/kotlin/net/corda/membership/impl/synchronisation/MemberSynchronisationServiceImpl.kt
Auto-merging components/membership/synchronisation-impl/src/test/kotlin/net/corda/membership/impl/synchronisation/MemberSynchronisationServiceImplTest.kt
Auto-merging components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRestResourceImpl.kt
CONFLICT (content): Merge conflict in components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRestResourceImpl.kt
Auto-merging gradle.properties
CONFLICT (content): Merge conflict in gradle.properties
Auto-merging libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepository.kt
CONFLICT (content): Merge conflict in libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepository.kt
Auto-merging libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepositoryImpl.kt
Auto-merging processors/member-processor/src/integrationTest/kotlin/net/corda/processor/member/TestMembershipPersistenceClientImpl.kt
Automatic merge failed; fix conflicts and then commit the result.
```

Patch (I updated PersistMemberInfoHandlerTest as a part of the merge commit, so it compiles even though the conflicts auto resolved):
```
+++ b/components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandler.kt
@@@ -5,7 -6,8 +6,9 @@@ import net.corda.data.KeyValuePairLis
  import net.corda.data.membership.db.request.MembershipRequestContext
  import net.corda.data.membership.db.request.command.PersistMemberInfo
  import net.corda.membership.datamodel.MemberInfoEntity
 +import net.corda.membership.lib.MemberInfoExtension.Companion.MEMBER_STATUS_PENDING
+ import net.corda.membership.datamodel.MemberInfoEntityPrimaryKey
+ import net.corda.membership.lib.MemberInfoExtension
  import net.corda.membership.lib.MemberInfoExtension.Companion.groupId
  import net.corda.membership.lib.MemberInfoExtension.Companion.status
  import net.corda.membership.lib.exceptions.MembershipPersistenceException
@@@ -32,6 -48,25 +49,29 @@@ internal class PersistMemberInfoHandler
              transaction(context.holdingIdentity.toCorda().shortHash) { em ->
                  request.members.forEach {
                      val memberInfo = memberInfoFactory.create(it)
+                     val currentMemberInfo = em.find(
+                         MemberInfoEntity::class.java,
 -                        MemberInfoEntityPrimaryKey(memberInfo.groupId, memberInfo.name.toString()),
++                        MemberInfoEntityPrimaryKey(
++                            memberInfo.groupId,
++                            memberInfo.name.toString(),
++                            memberInfo.status == MEMBER_STATUS_PENDING
++                        ),
+                         LockModeType.PESSIMISTIC_WRITE
+                     )
+                     if (currentMemberInfo?.serialNumber == memberInfo.serial) {
+                         val currentMemberContext = deserialize(currentMemberInfo.memberContext)
+                         val currentMgmContext = deserialize(currentMemberInfo.mgmContext)
+                         if (currentMemberContext.items != it.memberContext.items) {
+                             throw MembershipPersistenceException("Cannot update member info with same serial number " +
+                                 "(${memberInfo.serial}): member context differs from original.")
+                         }
+                         if (currentMgmContext.toMap().removeTime() != it.mgmContext.toMap().removeTime()) {
+                             throw MembershipPersistenceException("Cannot update member info with same serial number " +
+                                 "(${memberInfo.serial}): mgm context differs from original.")
+                         }
+                         return@forEach
+                     }
+
                      val entity = MemberInfoEntity(
                          memberInfo.groupId,
                          memberInfo.name.toString(),
diff --cc components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRestResourceImpl.kt
index 45aa3416ab,f6eefffa77..0000000000
--- a/components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRestResourceImpl.kt
+++ b/components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRestResourceImpl.kt
diff --cc gradle.properties
index 69b3dc8384,c0ac680879..0000000000
--- a/gradle.properties
+++ b/gradle.properties
@@@ -40,7 -40,7 +40,7 @@@ bouncycastleVersion=1.7
  # Corda API libs revision (change in 4th digit indicates a breaking change)
  # Change to 5.0.0.xx-SNAPSHOT to pick up maven local published copy
  #cordaApiVersion=5.0.0.xxx-SNAPSHOT
- cordaApiVersion=5.0.0.705-beta+
 -cordaApiVersion=5.0.0.664-Gecko-beta+
++cordaApiVersion=5.0.0.706-alpha-1678362967942

  disruptorVersion=3.4.2
  felixConfigAdminVersion=1.9.26
diff --cc libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepository.kt
index 7c684e12e6,9320f83739..0000000000
--- a/libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepository.kt
+++ b/libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/repository/VirtualNodeRepository.kt
diff --git a/components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandlerTest.kt b/components/members
hip/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandlerTest.kt
index 0419602045..98ae2a608e 100644
--- a/components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandlerTest.kt
+++ b/components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/PersistMemberInfoHandlerTest.kt
@@ -213,7 +213,11 @@ class PersistMemberInfoHandlerTest {
         }
         whenever(entityManager.find(
             eq(MemberInfoEntity::class.java),
-            eq(MemberInfoEntityPrimaryKey(requestContext.holdingIdentity.groupId, requestContext.holdingIdentity.x500Name.toString())),
+            eq(MemberInfoEntityPrimaryKey(
+                requestContext.holdingIdentity.groupId,
+                requestContext.holdingIdentity.x500Name.toString(),
+                false)
+            ),
             any<LockModeType>())
         ).doReturn (memberInfoEntity)

@@ -242,7 +246,11 @@ class PersistMemberInfoHandlerTest {
         }
         whenever(entityManager.find(
             eq(MemberInfoEntity::class.java),
-            eq(MemberInfoEntityPrimaryKey(requestContext.holdingIdentity.groupId, requestContext.holdingIdentity.x500Name.toString())),
+            eq(MemberInfoEntityPrimaryKey(
+                requestContext.holdingIdentity.groupId,
+                requestContext.holdingIdentity.x500Name.toString(),
+                false)
+            ),
             any<LockModeType>())
         ).doReturn (memberInfoEntity)

@@ -270,8 +278,11 @@ class PersistMemberInfoHandlerTest {
         }
         whenever(entityManager.find(
             eq(MemberInfoEntity::class.java),
-            eq(MemberInfoEntityPrimaryKey(requestContext.holdingIdentity.groupId, requestContext.holdingIdentity.x500Name.toString())),
-            any<LockModeType>())
+            eq(MemberInfoEntityPrimaryKey(
+                requestContext.holdingIdentity.groupId,
+                requestContext.holdingIdentity.x500Name.toString(),
+                false)
+            ),            any<LockModeType>())
         ).doReturn (memberInfoEntity)

         assertThrows<MembershipPersistenceException> {
@@ -308,7 +319,11 @@ class PersistMemberInfoHandlerTest {
         }
         whenever(entityManager.find(
             eq(MemberInfoEntity::class.java),
-            eq(MemberInfoEntityPrimaryKey(requestContext.holdingIdentity.groupId, requestContext.holdingIdentity.x500Name.toString())),
+            eq(MemberInfoEntityPrimaryKey(
+                requestContext.holdingIdentity.groupId,
+                requestContext.holdingIdentity.x500Name.toString(),
+                false)
+            ),
             any<LockModeType>())
         ).doReturn (memberInfoEntity)
```

I also reverted the Jenkins pipeline files:
```
git checkout  HEAD .ci/e2eTests/Jenkinsfile
Updated 1 path from 202979d303
git checkout HEAD .ci/nightly/JenkinsfileNightly
Updated 1 path from 202979d303
git checkout  HEAD Jenkinsfile
Updated 1 path from 202979d303
git checkout  HEAD buildSrc/src/main/groovy/DeployableContainerBuilder.groovy
Updated 1 path from 202979d303
git checkout HEAD charts/corda/Chart.yaml
Updated 1 path from 202979d303
git checkout HEAD tools/plugins/build.gradle
Updated 1 path from 202979d303
```
and reverted some changes to gradle.properties changes:
```
diff --git a/gradle.properties b/gradle.properties
index 7360ce1015..e20092ee44 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -15,7 +15,7 @@ platformVersion = 999
 cordaProductVersion=5.0.0
 ## The revision number. This lines up the versioning of the runtime-os repo with the API repo, which allows the build
 ## system to assume the same versioning scheme.
-cordaRuntimeRevision=0-Gecko
+cordaRuntimeRevision=0

 # Plugin dependency versions
 bndVersion=6.4.0
@@ -76,7 +76,7 @@ osgiUtilFunctionVersion = 1.2.0
 osgiUtilPromiseVersion = 1.3.0
 picocliVersion = 4.7.0
 protonjVersion=0.33.0
-quasarVersion = 0.8.10_r3-20230307.105917-12
+quasarVersion = 0.8.10_r3-SNAPSHOT
 reflectAsmVersion = 1.11.9
 slf4jVersion=1.7.36
 snappyVersion=0.4
@@ -122,7 +122,7 @@ artifactoryPluginVersion = 4.28.2
 pf4jVersion=3.8.0

 # corda-cli plugin host
-pluginHostVersion=5.0.0-Gecko-beta+
+pluginHostVersion=5.0.0-beta+
 systemLambdaVersion=1.2.1

 # DB integration tests
```

### CODE REVIEW

**Review by dimosr — DISMISSED** [review]
LGTM, but worth getting a review from @chrisr3 / @ronanbrowne for their changes as well as I don't have the context on the changes around Quasar and the Jenkins files.

**Comment by williamvigorr3** [comment]
API PR: https://github.com/corda/corda-api/pull/927

**Review by mcgovc — DISMISSED** [review]
LGTM - build changes

File: gradle.properties [hunk 1]
```diff
@@ -40,7 +40,7 @@ bouncycastleVersion=1.72
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.0.0.xx-SNAPSHOT to pick up maven local published copy
 #cordaApiVersion=5.0.0.xxx-SNAPSHOT
-cordaApiVersion=5.0.0.705-beta+
+cordaApiVersion=5.0.0.706-alpha-1678362967942
```
Comment (chrisr3) [inline gradle.properties]: Not sure about this bit?! :thinking:

**Review by chrisr3 — COMMENTED** [review]
LGTM, in the sense that `release/os/5.0` needs to continue using Quasar 0.8.10_r3-SNAPSHOT :+1:.

File: gradle.properties (hunk 1 above)
Comment (williamvigorr3) [reply gradle.properties]: I will revert this. Going to take the api lock.

Comment (williamvigorr3) [reply gradle.properties]: By revert I mean it will become `cordaApiVersion=5.0.0.706-beta+` once https://github.com/corda/corda-api/pull/927 is merged.

**Review by dimosr — APPROVED** [review]
re-approving assuming it's only the api bump (it would be nice to have a single commit indicating that without the force-push)

**Comment by williamvigorr3** [comment]
> re-approving assuming it's only the api bump (it would be nice to have a single commit indicating that without the force-push)

I forced pushed because otherwise we will end up with another commit in the history (as this isn't being squashed). There is a compare button. It was only the api bump.

**Comment by dimosr** [comment]
ah nice, I didn't know about the compare button. That's very handy

**Pull request merged**